Start using daohaus metadata when populating DAO information

alalonde opened the pull request:
## Overview

Changes the daohaus lookup code to fetch DAO metadata from an undocumented daohaus "API" 

Closes #646, #838

## Follow up Improvement Ideas

- Cache this data appropriately.  This makes the DAO lookup even slower since we're doing an additional HTTP request for each individual DAO. (See #997)
- Implement backend and UI for defining a whitelist of DAOs you want to include in your profile

## Assets

![daohaus](https://user-images.githubusercontent.com/1402582/154547100-2a968c7f-758b-43a6-8827-10f57699711e.png)

alalonde commented:
vercel is failing due to the backend schema changing.  Not sure why the GCD action didn't deploy...

dan13ram commented:
This looks so cool. May not be necessary in this PR, but we should add a tag to each DAO to display which network it is on?
The same for NFTs too would be cool (once we start displaying NFTs from other networks as well)

dan13ram reviewed (approved):
Great work! 

I see that the svg chain icons aren't being used. I think we could start using them instead of the pngs. Otherwise everything else is 👌🏽

dan13ram commented on packages/web/utils/daoHelpers.ts:
```diff
@@ -0,0 +1,20 @@
+  if (address && chain) {
+    const hexChainId = getHexChainId(chain);
+    address = address.toLowerCase(); // eslint-disable-line no-param-reassign
+    return `https://app.daohaus.club/dao/${hexChainId}/${address}`;
```
actually the method can be shortened to 
```
export const getDaoLink = (chain?: string, address?: string): string | null =>
  address && chain
    ? `https://app.daohaus.club/dao/${getHexChainId(chain)}/${address.toLowerCase()}`
    : null;
```

alalonde commented on packages/design-system/src/icons/ChainIcon.tsx:
```diff
@@ -0,0 +1,17 @@
+  chain: string | undefined;
+};
+
+export const ChainIcon: React.FC<Props & IconProps> = ({ chain, ...props }) => {
```
Ha, yes, I added this to be able to show a little chain icon next to the DAO as well, but the various svgs rendered weird so I bailed on the idea.

alalonde commented:
Alright, just made another rather large update. I removed the old chain images in favor of the new SVG one, and tweaked the SVGs so that they render better. I also tested and updated the /player/setup/memberships component to match the one for viewing players

dan13ram commented on packages/web/components/Player/PlayerGuild.tsx:
```diff
@@ -1,78 +1,56 @@
+  children,
+}) => (
+  <Link
+    role="group"
```
Why is there a role=group here ? I think its creating some issues.

dan13ram replied:
![CPT2202191414-472x522](https://user-images.githubusercontent.com/10012209/154793771-87dde509-191f-4574-ac9a-570b4dea21e5.gif)

alalonde replied:
Well, all I did was split these components into three discrete components without changing their functionality. So `role="group"` has been there all along.

dan13ram replied:
oh right. still i think it shouldn't be there

alalonde replied:
_groupHover is used in PlayerGuildMemberships as one of the `children`

dan13ram replied:
Are you seeing those UI bugs on hover or its misbehaving only on my side?

alalonde replied:
I see it on Chrome but not on FF

dan13ram commented on packages/web/components/Player/PlayerGuild.tsx:
```diff
@@ -1,78 +1,56 @@
+export const DaoHausLink: React.FC<DaoHausLinkProps> = ({ daoUrl, children }) =>
+  daoUrl != null ? (
+    <Link
+      role="group"
```
Here as well.
